tests/test_server.py

```diff
@@ -238,6 +238,72 @@ def test_delete_nonexistent_product_from_shopcart(self):
         self.assertEqual( resp.status_code, status.HTTP_204_NO_CONTENT )
         cart = server.Shopcart.find(2)
         self.assertEqual( (5 in cart.products), False)
+    
+    def test_add_new_product_to_cart(self):
+        """ Add a new product to the cart """
+        product_count = len(server.Shopcart.find(2).products)
+        data = {1: 2}
+        
+        resp = self.app.post('/shopcarts/2/products', data=json.dumps(data), content_type='application/json')
+        data = json.loads(resp.data.decode('utf8'))
+        
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(data['products']), product_count + 1)
+        self.assertEqual(data['products']['1'], 2)
+    
+    def test_add_multiple_new_products_to_cart(self):
+        """ Add multiple new products to the cart """
+        product_count = len(server.Shopcart.find(2).products)
+        data = {1: 2, 2: 4}
+        
+        resp = self.app.post('/shopcarts/2/products', data=json.dumps(data), content_type='application/json')
+        data = json.loads(resp.data.decode('utf8'))
+        
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(data['products']), product_count + 2)
+        self.assertEqual(data['products']['1'], 2)
+        self.assertEqual(data['products']['2'], 4)
+    
+    def test_add_existing_product_to_cart(self):
+        """ Add an existing product to the cart """
+        product_count = len(server.Shopcart.find(4).products)
+        data = {5: 2}
+        
+        resp = self.app.post('/shopcarts/4/products', data=json.dumps(data), content_type='application/json')
+        data = json.loads(resp.data.decode('utf8'))
+        
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(data['products']), product_count)
+        self.assertEqual(data['products']['5'], 9)
```

Comment:
It looks good but mayybe we have to be extra consistent .
Shouldn't we do different types of requests when we have an existing product so that it becomes a 

POST  to /shopcart/uid/products only when the cart does not have the existing pid to add 
and 
PUT to /shopcart/uid/products/pid  to add to the pid?? @thoughts1053 @radhikamattoo

Reply:
But we also have the update funciton  with  PUT  /shopcart/uid/products/pid, so I don't know. Maybe overthinking...

Reply:
I'm fine with how it is, it's just me trying to clear my mind haha

Reply:
Whether the pid exists or not is abstracted away from the router - you can see in the handler here: https://github.com/nyu-devops-echo/shopcarts/blob/b47caf36d7dff5a626849a2bfa232b36e72a482e/server.py#L125
that we never actually handle that on the router side - we simply get the `POST` data, and ask the model to do the work for us.

Reply:
OK OK! I'm actually ok with it just trying to understand REST because its confusing for me.

Reply:
Yeah, this was a little different from standard rest because we have 2 scenarios for 'updating' a product. The `PUT` will basically update the quantity with whatever you pass in. Whereas this `POST` will add the quantity to your shopping cart. So if it already existed we want to add to the quantity instead of replacing it.